core/vm/contracts.go

```diff
@@ -1379,36 +1389,49 @@ func kZGToVersionedHash(kzg kzg4844.Commitment) common.Hash {
 	return h
 }
 
+// P256VERIFY (secp256r1 signature verification)
+// implemented as a native contract.
+//
+// This is used in the OP Stack from Fjord until the implementation of Ethereum's Osaka fork, after
+// which the p256Verify precompile is used instead to maintain Ethereum equivalence.
+type p256VerifyFjord struct {
+	inner p256Verify
+}
+
+// RequiredGas returns the gas required to execute the precompiled contract
+func (c *p256VerifyFjord) RequiredGas(input []byte) uint64 {
+	return params.P256VerifyGas
+}
+
+// Run executes the precompiled contract with given 160 bytes of param, returning the output and the used gas
+func (c *p256VerifyFjord) Run(input []byte) ([]byte, error) {
+	return c.inner.Run(input)
+}
```

Comment:
Any reason not to use an embed? Then `Run` doesn't need to be specified again.
```suggestion
type p256VerifyFjord struct {
	p256Verify
}

// RequiredGas returns the gas required to execute the precompiled contract
func (c *p256VerifyFjord) RequiredGas(input []byte) uint64 {
	return params.P256VerifyGasFjord // if we rename
}
```